cli/cmd/deploy/deploy.go

```diff
@@ -322,17 +322,33 @@ func verifyAccess(ctx context.Context, c *client.Client, githubURL string) (*adm
 		GithubUrl: githubURL,
 	})
 	if err != nil {
+		if st, ok := status.FromError(err); ok && st.Code() == codes.PermissionDenied {
+			fmt.Println(st.Message())
+		}
 		return nil, err
 	}
 
+	userAuthRequested := false
 	// If the user has not already granted access, open browser and poll for access
 	if !ghRes.HasAccess {
-		// Print instructions to grant access
-		fmt.Printf("Rill projects deploy continuously when you push changes to Github.\n")
-		fmt.Printf("You need to install rill github app to grant read only access to your project.\n\n")
-		time.Sleep(3 * time.Second)
-		fmt.Printf("Open this URL in your browser to grant Rill access to your Github repository:\n\n")
-		fmt.Printf("\t%s\n\n", ghRes.GrantAccessUrl)
+		if ghRes.GrantAccessUrl != "" {
```

Comment:
Not a fan of this conditional – see comment on proto file for how we could rid of it